src/PayID/pay-id-client.ts

```diff
@@ -246,27 +192,31 @@ export default class PayIDClient implements PayIDClientInterface {
 
     const apiInstance = new DefaultApi(client)
 
-    return new Promise((resolve, reject) => {
-      try {
-        apiInstance.postPathInvoice(
-          { body: signatureWrapper },
-          (error, data, _response) => {
-            // TODO(keefertaylor): Provide more granular error handling.
-            if (error) {
-              const message = `${error.status}: ${error.response?.text}`
-              reject(new PayIDError(PayIDErrorType.UnexpectedResponse, message))
-            } else if (data) {
-              resolve(data)
-            } else {
-              reject(new PayIDError(PayIDErrorType.UnexpectedResponse))
-            }
-          },
-        )
-      } catch (exception) {
-        // Something really wrong happened, we don't have enough information to tell. This could be a transient network error, the payment pointer doesn't exist, or any other number of errors.
-        reject(new PayIDError(PayIDErrorType.Unknown, exception.message))
-      }
+    const { error, data } = await new Promise<
+      SwaggerCallResult<SignatureWrapperInvoice>
+    >((resolve, _reject) => {
+      apiInstance.postPathInvoice(
+        { body: signatureWrapper },
+        (swaggerError, swaggerData, response) => {
+          // Transform results of the callback to a wrapper object.
+          resolve({
+            error: swaggerError,
+            data: swaggerData,
+            response,
+          })
+        },
+      )
     })
+
+    // TODO(keefertaylor): Provide more granular error handling.
+    if (error) {
+      const message = `${error.status}: ${error.response?.text}`
+      throw new PayIDError(PayIDErrorType.UnexpectedResponse, message)
```

Comment:
Wondering -- why not return the error and error type returned by the PayID server on all of these?

I would think that the services using the SDK would want to receive the error messages from the PayID server, but was curious as to what your thinking was around this.

Reply:
I generally favor using enum based errors because it limits the ranges of errors a client has to handle and makes code more predictable. It's possible (and maybe we should) to include the underlying error here if we really wanted to. 

I suspect / know that error handling is going to change a fair deal so I've just done this for now because it works for the moment.